src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -531,6 +524,119 @@ private void AddSchema(CsdlSchema schema, bool addAnnotations)
+            {
+                AddOutOfLineAnnotationsFromSchema(schema, schemaWrapper, includeAnnotationsIndex: null);
+            }
+
+            var edmVersion = this.GetEdmVersion();
+            if (edmVersion == null || edmVersion < schema.Version)
+            {
+                this.SetEdmVersion(schema.Version);
+            }
+        }
+
+        private void AddSchemaIfReferenced(CsdlSchema schema, IEnumerable<IEdmReference> parentReferences)
+        {
+            CsdlSemanticsSchema schemaWrapper = new CsdlSemanticsSchema(this, schema);
+
+            bool shouldAddSchemaElements;
+            Dictionary<string, List<IEdmIncludeAnnotations>> includeAnnotationsIndex;
+            ProcessSchemaParentReferences(schema, schemaWrapper, parentReferences, out shouldAddSchemaElements, out includeAnnotationsIndex);
+
+            if (!string.IsNullOrEmpty(schema.Alias))
+            {
+                this.SetNamespaceAlias(schema.Namespace, schema.Alias);
+            }
+
+            if (shouldAddSchemaElements)
+            {
+                AddSchemaElements(schema, schemaWrapper);
+            }
+
+            if (includeAnnotationsIndex.Count > 0)
+            {
+                AddOutOfLineAnnotationsFromSchema(schema, schemaWrapper, includeAnnotationsIndex);
+            }
+
+            if (shouldAddSchemaElements || includeAnnotationsIndex.Count > 0)
+            {
+                this.schemata.Add(schemaWrapper);
+
+                var edmVersion = this.GetEdmVersion();
+                if (edmVersion == null || edmVersion < schema.Version)
+                {
+                    this.SetEdmVersion(schema.Version);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Process the <see cref="IEdmReference"/>s of the parent model to
+        /// figure out whether the schema's types (entiy types, enums, containers, etc.) and annotations should
+        /// be imported
+        /// </summary>
+        /// <param name="schema"></param>
+        /// <param name="schemaWrapper"></param>
+        /// <param name="parentReferences"></param>
+        /// <param name="shouldAddSchemaElements">Whether schema types such entity types, operations, enums, etc. should be important</param>
+        /// <param name="includeAnnotationsIndex">Cache of the schema's out-of-line <see cref="IEdmIncludeAnnotations"/>s indexed by the term namespace</param>
+        private static void ProcessSchemaParentReferences(
+            CsdlSchema schema,
+            CsdlSemanticsSchema schemaWrapper,
+            IEnumerable<IEdmReference> parentReferences,
+            out bool shouldAddSchemaElements,
+            out Dictionary<string, List<IEdmIncludeAnnotations>> includeAnnotationsIndex)
+        {
+            string schemaNamespace = schema.Namespace;
+
+            shouldAddSchemaElements = false;
+            includeAnnotationsIndex = new Dictionary<string, List<IEdmIncludeAnnotations>>();

```

Comment:
Maybe apply the constructor that takes an `IEqualityComparer<T>` argument if `TermNamespace="Org.OData.Capabilities.V1"` is considered to be equivalent to `TermNamespace="org.OData.Capabilities.V1"` (i.e. case-insensitive)

Reply:
I believe namespaces are case-sensitive. At least in many cases, the spec mentions that identifiers are case-sensitive. I assume it applies to namespaces as well.